spec/controllers/api/v1/books_controller_spec.rb

```diff
@@ -80,4 +80,93 @@
       end
     end
   end
+
+  describe 'GET #fetch_book_by_isbn' do
+    context 'when using a valid isbn' do
+      let(:expected_response) do
+        {
+          'isbn': '9780876855577',
+          'title': 'Ham on rye',
+          'subtitle': 'a novel',
+          'number_of_pages': 283,
+          'authors': ['Charles Bukowski']
+        }
+      end
+
+      before do
+        body = {
+          'ISBN:9780876855577': {
+            'subtitle': 'a novel',
+            'title': 'Ham on rye',
+            'number_of_pages': 283,
+            'authors': [{
+              'url': 'https://openlibrary.org/authors/OL31084A/Charles_Bukowski',
+              'name': 'Charles Bukowski'
+            }]
+          }
+        }
+        stub_request(:get, 'https://openlibrary.org/api/books?bibkeys=ISBN:9780876855577&format=json&jscmd=data')
+          .to_return(
+            headers: { 'Content-Type': 'application/json' },
+            body: body.to_json
+          )
```

Comment:
just so you know, you can put this stubs in a separate folder for mocks and stubs. That way you would have a cleaner test and reusable stubs